### PR TITLE
Dataflow sensors - job metrics

### PR DESCRIPTION
**Pull request by TobKed**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by kaxil** [comment]
Can you please rebased your PR on latest Master since we have applied [Black](https://github.com/apache/airflow/commit/4e8f9cc8d02b29c325b8a5a76b4837671bdf5f68) and [PyUpgrade](https://github.com/apache/airflow/commit/8c42cf1b00c90f0d7f11b8a3a455381de8e003c5) on Master

File: airflow/providers/google/cloud/sensors/dataflow.py [hunk 1]
```diff
@@ -113,3 +113,75 @@ def poke(self, context: dict) -> bool:
...
+    """
+
+    template_fields = ['job_id']
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        job_id: str,
+        callback: Callable,
```
Comment (turbaszek) [inline airflow/providers/google/cloud/sensors/dataflow.py]: Have you check how well it works with serialisation? Serialising a function can be problematic.cc @kaxil @mik-laj

Comment (TobKed) [reply airflow/providers/google/cloud/sensors/dataflow.py]: @turbaszek could you provide some more information how to test it for serialization, please? I am not so familiar with this topic

Comment (mik-laj) [reply airflow/providers/google/cloud/sensors/dataflow.py]: it is safe for serialization.

**Comment by TobKed** [comment]
PTAL @aaltay @mik-laj @turbaszek

**Review by aaltay — APPROVED** [review]
Nice!

**Review by mik-laj — APPROVED** [review]

**Pull request merged**